sdk/identity/azure-identity/azure/identity/aio/_authn_client.py

```diff
@@ -61,6 +62,23 @@ def __init__(
         token = self._deserialize_and_cache_token(response=response, scopes=scopes, request_time=request_time)
         return token
 
+    async def obtain_token_by_refresh_token(self, scopes: Iterable[str]) -> Optional[AccessToken]:
+        """Acquire an access token using a cached refresh token. Returns ``None`` when that fails, or the cache has no
+        refresh token. This is only used by SharedTokenCacheCredential and isn't robust enough for anything else."""
+
+        # these tokens come from the user-specific shared token cache, so all belong to the current user
+        for token in self.get_refresh_tokens(scopes):
```

Comment:
As we discussed it would be good to look for a specific Username here rather than the first that returns a token.